jujuclient/controllers.go

```diff
@@ -7,15 +7,28 @@ import (
 	"github.com/juju/errors"
 )
 
+const controllersLockName = "controllers.lock"
+
 // AllControllers implements ControllersGetter.AllControllers.
 // This implementation gets all controllers defined in the controllers file.
 func (f *store) AllControllers() (map[string]ControllerDetails, error) {
+	lock, err := acquireEnvironmentLock(controllersLockName, "read-all-controllers")
```

Comment:
a bit verbose; can you please add a `store.lock` method (and unlock) so we stop repeating things?